internal/abci/abci.go

```diff
@@ -685,14 +681,71 @@ func (a *AbciApp) prepareMempoolTxns(txs [][]byte, maxBytes int, log *log.Logger
 
 		if bytes.Equal(tx.Sender, proposerAddr) {
 			proposerNonce = tx.Body.Nonce + 1
-		}
+			propTxs = append(propTxs, tx)
+		} else {
+			// Append the transaction to the final list.
+			if _, ok := otherTxns[string(tx.Sender)]; !ok {
+				otherTxns[string(tx.Sender)] = make([]*indexedTxn, 0)
+				senders = append(senders, string(tx.Sender))
 
-		finalTxns = append(finalTxns, txs[tx.is])
+			}
+			otherTxns[string(tx.Sender)] = append(otherTxns[string(tx.Sender)], tx)
```

Comment:
Every single `string(tx.Sender)` makes it copy the bytes.  We can do it once when we enter the `else`